diskmaker/controllers/lv/reconcile.go

```diff
@@ -345,8 +340,8 @@ func (r *LocalVolumeReconciler) Reconcile(ctx context.Context, request ctrl.Requ
 
 	err = os.MkdirAll(r.symlinkLocation, 0755)
 	if err != nil {
-		logger.Error(err, " error creating local-storage directory ", "symLinkLocation", r.symlinkLocation)
-		os.Exit(-1)
+		klog.ErrorS(err, "error creating local-storage directory", "symLinkLocation", r.symlinkLocation)
+		return ctrl.Result{}, err
```

Comment:
This is buried in the logging changes, so I just want to point out the `os.Exit` call I removed here--that seemed very wrong to me.

Reply:
I distinctly remember spotting this during review, but at that time my thinking was - if we can't create local-storage root directory, something is indeed wrong with the configuration/node and exiting here will cause pod to crash loop and throw easy to spot errors that might be helpful (basically complain as loudly as we can, if things aren't really working).